src/routeGeneration/templates/hapi.hbs

```diff
@@ -77,9 +78,14 @@ export function RegisterRoutes(server: any) {
                     try {
                         validatedArgs = getValidatedArgs(args, request);
                     } catch (err) {
-                        return h
-                            .response(err)
-                            .code(err.status || 500);
+                        const boomErr = boomify(err);
+                        boomErr.output.statusCode = err.status || 500;
+                        boomErr.output.payload = {
+                            name: err.name,
+                            fields: err.fields,
+                            message: err.message,
+                        } as unknown as Payload;
```

Comment:
The `@hapi/boom` [Payload type](https://github.com/hapijs/boom/blob/7e94679c2111ee49f88a787499a36aedba207c83/lib/index.d.ts#L93-L108) is incorrect, which is why we need to force cast here.

The [docs here](https://github.com/hapijs/hapi/blob/fa76016359245c9f46a2332fd8bab82ae3b5bba9/API.md#error-transformation) say you can make payload anything you want, so the typescript type should have all optional `?` keys and should allow extra keys

Reply:
Can you try to set this during the initial call to `boomify()`: `boomify(err, {decorate: {fields, message}})`

Reply:
Yes I gave that a try and unfortunately it doesn't work. From the docs:
```
decorate - an option with extra properties to set on the error object.
```

So `decorate` adds fields to the JS error object, but those fields don't end up in the payload response from the server